test/index.spec.js

```diff
@@ -0,0 +1,60 @@
+import chai from 'chai';
+import configureStore from 'redux-mock-store';
+import promiseMiddleware from '../src/index';
+
+describe('promise middleware', () => {
+  const nextHandler = promiseMiddleware()();
+
+  it('must return a function to handle next', () => {
+    chai.assert.isFunction(nextHandler);
+    chai.assert.strictEqual(nextHandler.length, 1);
+  });
+
+  describe('handle next', () => {
+    it('must return a function to handle action', () => {
+      const actionHandler = nextHandler();
+      chai.assert.isFunction(actionHandler);
+    });
+  });
+
+  describe('handle action', () => {
+    const middlewares = [promiseMiddleware()];
+    const mockStore = configureStore(middlewares);
+    const mockActionWithoutPromise = { type: 'GET_POST' };
+    const mockRejectedAction = {
+      type: 'GET_POST',
+      payload: {
+        promise: () => {
+          return new Promise((resolve, reject) => {
+            reject(new Error());
+          });
+        }
+      }
+    };
+    const mockFulfilledAction = {
+      type: 'GET_POST',
+      payload: {
+        promise: () => {
+          return new Promise((resolve, reject) => {
+            resolve({});
+          })
+        }
+      }
+    };
+
+    it('must pass action to next if not a promise', done => {
+      const store = mockStore({}, [mockActionWithoutPromise], done);
+      store.dispatch(mockActionWithoutPromise);
+    });
+
+    it('must dispatch rejected action if promise is rejected', done => {
+      const store = mockStore({}, [mockRejectedAction], done);
```

Comment:
I don't understand this, it's testing that the action is dispatched with exactly the same type and also including the same payload, shouldn't we be testing that it's dispatched our three suffix types?

```
    const first = {
      type: 'GET_POST_PENDING'
    };
    const rejected = {
      type: 'GET_POST_REJECTED',
      error: true
    };
    const fulfilled = {
      type: 'GET_POST_FULFILLED',
      payload: {}
    };
```